src/sentry/utils/appleconnect/appstore_connect.py

```diff
@@ -214,19 +190,20 @@ def get_apps(session: Session, credentials: AppConnectCredentials) -> Optional[L
     url = "v1/apps"
     ret_val = []
     try:
-        apps = _get_appstore_info_paged_data(session, credentials, url)
-        for app in apps:
-            app_info = AppInfo(
-                app_id=app.get("id"),
-                bundle_id=safe.get_path(app, "attributes", "bundleId"),
-                name=safe.get_path(app, "attributes", "name"),
-            )
-            if (
-                app_info.app_id is not None
-                and app_info.bundle_id is not None
-                and app_info.name is not None
-            ):
-                ret_val.append(app_info)
+        app_pages = _get_appstore_info_paged(session, credentials, url)
+        for app_page in app_pages:
+            for app in app_page["data"]:
+                app_info = AppInfo(
+                    app_id=app.get("id"),
+                    bundle_id=safe.get_path(app, "attributes", "bundleId"),
+                    name=safe.get_path(app, "attributes", "name"),
+                )
+                if (
+                    app_info.app_id is not None
+                    and app_info.bundle_id is not None
+                    and app_info.name is not None
+                ):
```

Comment:
Hmm, apparently we already did this silently skipping of malformed data... I'd really be more comfortable with a sentry error here.  Maybe it's not fine to entirely throw an exception as returning as much as possible should be sane, but manually capturing the error might still be worth it (and if we find in practice this never triggers maybe raising an exception is still an option later)